lib/component.cpp

```diff
@@ -1,10 +1,85 @@
 #include "component.hpp"
+#include "logical_address.hpp"
+#include "messages/spa_subscription_reply.hpp"
+#include "messages/spa_subscription_request.hpp"
+#include <iostream>
+#include <memory>
 
-void Component::sendMsg(std::shared_ptr<SpaMessage> message)
+void Component::registerSubscriptionRequest(std::shared_ptr<SpaMessage> message)
 {
-  if (message == nullptr || communicator == nullptr)
+  if (message == nullptr)
+    return;
+
+  auto castMessage = std::dynamic_pointer_cast<SpaSubscriptionRequest>(message);
+
+  auto reply = std::make_shared<SpaSubscriptionReply>(
+      0,                            // Version
+      0,                            // Message priority
+      address,                      // Address of the producer component
+      castMessage->consumerAddress, // Address of the producer component
+      castMessage->dialogId,        // Dialog identifer sent by requester
+      0                             // 0 = accepted
+      );
+
+  sendMsg(reply);
+}
+
+void Component::handleSubscriptionReply(std::shared_ptr<SpaMessage> message)
+{
+  if (message == nullptr)
+    return;
+  auto castMessage = std::dynamic_pointer_cast<SpaSubscriptionReply>(message);
+
+  if (castMessage->replyType != 0)
+  {
+    std::cout << "Subscription failed." << std::endl;
```

Comment:
Maybe add a TODO in here so that someone else will know that logic needs to be added to keep track of the subscriptions